apps/internal/oauth/ops/authority/authority.go

```diff
@@ -213,9 +214,23 @@ func NewInfoFromAuthorityURI(authorityURI string, validateAuthority bool) (Info,
 		return Info{}, err
 	}
 
+	port := u.Port()
+	hostname := u.Hostname()
+	cannonicalAuthorityURI = fmt.Sprintf("https://%v/%v/", u.Hostname(), tenant)
+	if port != "" {
+		hostname = fmt.Sprintf("%v:%v", hostname, port)
+		cannonicalAuthorityURI = fmt.Sprintf("https://%v/%v/", hostname, tenant)
+	}
+	// Note: temporary workaround to support private cloud scenarios while the following GitHub issues are addressed
+	// https://github.com/AzureAD/microsoft-authentication-library-for-go/issues/301
+	// https://github.com/AzureAD/microsoft-authentication-library-for-go/issues/289
+	disableAuthorityValidation, ok := os.LookupEnv("DISABLE_AUTHORITY_VALIDATION")
+	if ok {
+		validateAuthority = !(disableAuthorityValidation == "1")
```

Comment:
If the user has already set `WithInstanceDiscovery(false)` won't this option override it?